### PR TITLE
Adding a Delete method

### PR DESCRIPTION
**Pull request by dangermike**
Delete removes an item from the cache. There is no effect if the item isn't there. If it is there, the item is removed from both the map (search index) and the heap (expiration index).

### CODE REVIEW

File: generic/lazylru.go [hunk 1]
```diff
@@ -397,6 +397,26 @@ func (lru *LazyLRU[K, V]) MSetTTL(keys []K, values []V, ttl time.Duration) error
...
+	lru.lock.RLock()
+	_, ok := lru.index[key]
+	lru.lock.RUnlock()
+	if !ok {
+		return
+	}
+	lru.lock.Lock()
+	pqi, ok := lru.index[key]
+	if !ok {
+		return
```
Comment (aldelucca1) [inline generic/lazylru.go:413]: This branch needs to unlock the mutex

File: lazylru.go [hunk 2]
```diff
@@ -375,6 +375,26 @@ func (lru *LazyLRU) MSetTTL(keys []string, values []interface{}, ttl time.Durati
...
+	lru.lock.RLock()
+	_, ok := lru.index[key]
+	lru.lock.RUnlock()
+	if !ok {
+		return
+	}
+	lru.lock.Lock()
+	pqi, ok := lru.index[key]
+	if !ok {
+		return
```
Comment (aldelucca1) [inline lazylru.go:391]: Same here regarding the unlock

File: generic/lazylru.go (hunk 1 above)
Comment (dangermike) [reply generic/lazylru.go:413]: oops! thanks!

Comment (dangermike) [reply generic/lazylru.go:413]: that would have taken forever to find -- lock up the whole process but only if there is a race condition that you lose when deleting an item multiple times. oof!

File: lazylru.go (hunk 2 above)
Comment (dangermike) [reply lazylru.go:391]: updated

**Review by aldelucca1 — APPROVED** [review]

**Pull request merged**